routes/publishedSchedule.js

```diff
@@ -92,21 +124,57 @@ publishedScheduleRouter.get('/date', async (req, res) => {
     const { date } = req.query;
     const seasonResult = await db.query(
       `
-      SELECT
-        PS.id,
-        C.title,
-        C.event_type,
-        C.year,
-        PS.start_time,
-        PS.end_time,
-        PS.confirmed,
-        PS.confirmed_on,
-        PS.cohort,
-        PS.notes
-      FROM published_schedule PS
-      LEFT JOIN catalog C ON PS.event_id = C.id
-      WHERE DATE(PS.start_time) = $1
-      ORDER BY start_time ASC;
+      WITH seasonPS AS
+      (
+        SELECT
+          PS.id,
+          PS.day_id,
+          D.event_date,
+          D.start_time AS day_start_time,
+          D.end_time AS day_end_time,
+          D.location,
+          D.notes AS day_notes,
+          C.title,
+          C.event_type,
+          C.year,
+          PS.start_time,
+          PS.end_time,
+          PS.confirmed,
+          PS.confirmed_on,
+          PS.cohort,
+          PS.notes
+        FROM published_schedule PS
+        LEFT JOIN catalog C ON PS.event_id = C.id
+        LEFT JOIN day D on PS.day_id = D.id
+        WHERE
+          DATE(PS.start_time) = $1::date
+      )
+
+      SELECT json_build_object (
+        'id', seasonPS.day_id,
+        'event_date', seasonPS.event_date,
+        'start_time', seasonPS.day_start_time,
+        'end_time', seasonPS.day_end_time,
+        'location', seasonPS.location,
+        'notes', seasonPS.day_notes
+      ) as day,
+      JSON_AGG(
+        json_build_object (
+          'id', seasonPS.id,
+          'title', seasonPS.title,
+          'event_type', seasonPS.event_type,
+          'year', seasonPS.year,
+          'start_time', seasonPS.start_time,
+          'end_time', seasonPS.end_time,
+          'confirmed', seasonPS.confirmed,
+          'confirmed_on', seasonPS.confirmed_on,
+          'cohort', seasonPS.cohort,
+          'notes', seasonPS.notes
+        )
+      ) AS data
+      FROM seasonPS
+      GROUP BY DATE(start_time), day_id, event_date, day_start_time, day_end_time, location, day_notes 
+      ORDER BY DATE(start_time) ASC;
```

Comment:
for this query, it might be easier to filter by the date in the day table then join it with the referenced day_id.